File: app/schemas/knowledge.py
from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeDocumentResponse(BaseModel):
    id: str
    title: str
    source: str
    source_type: str
    version: str | None = None
    published_at: datetime | None = None
    checksum: str
    status: str
    chunk_count: int
    created_at: datetime
    updated_at: datetime


class KnowledgeStatusResponse(BaseModel):
    rag_enabled: bool
    rag_required: bool
    rag_ready: bool
    milvus: str
    collection: str
    embedding_model: str | None
    knowledge_documents: int = 0
    message: str | None = None


class KnowledgeDocumentPage(BaseModel):
    items: list[KnowledgeDocumentResponse] = Field(default_factory=list)
    page: int
    page_size: int
    total: int

File: web/src/api/knowledge.ts
import { request } from '../utils/request'

export const ragEnabled = import.meta.env.VITE_FEATURE_RAG !== 'false'

export interface KnowledgeStatus {
  rag_enabled: boolean
  rag_required: boolean
  rag_ready: boolean
  milvus: string
  collection: string
  embedding_model: string | null
  knowledge_documents: number
  message: string | null
}

export interface KnowledgeDocument {
  id: string
  title: string
  source: string
  source_type: string
  version: string | null
  published_at: string | null
  checksum: string
  status: string
  chunk_count: number
  created_at: string
  updated_at: string
}

export async function getKnowledgeStatus(): Promise<KnowledgeStatus> {
  return (await request.get<KnowledgeStatus>('/knowledge/status')).data
}

export async function getKnowledgeDocuments(): Promise<{ items: KnowledgeDocument[]; page: number; page_size: number; total: number }> {
  return (await request.get('/knowledge/documents')).data
}
